File: app/src/main/java/com/mytlogos/enterprise/background/room/model/RoomTocStat.java
package com.mytlogos.enterprise.background.room.model;

import androidx.annotation.NonNull;

public class RoomTocStat {
    private final int mediumId;
    private final int tocCount;

    public RoomTocStat(int mediumId, int tocCount) {
        this.mediumId = mediumId;
        this.tocCount = tocCount;
    }

    public int getMediumId() {
        return mediumId;
    }

    public int getTocCount() {
        return tocCount;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        RoomTocStat that = (RoomTocStat) o;

        if (getMediumId() != that.getMediumId()) return false;
        return getTocCount() == that.getTocCount();
    }

    @Override
    public int hashCode() {
        int result = getMediumId();
        result = 31 * result + getTocCount();
        return result;
    }

    @NonNull
    @Override
    public String toString() {
        return "RoomTocStat{" +
                "mediumId=" + mediumId +
                ", tocCount=" + tocCount +
                '}';
    }
}

File: app/src/main/java/com/mytlogos/enterprise/background/room/model/RoomTocStat.kt
package com.mytlogos.enterprise.background.room.model

class RoomTocStat(val mediumId: Int, val tocCount: Int) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as RoomTocStat
        return if (mediumId != that.mediumId) false else tocCount == that.tocCount
    }

    override fun hashCode(): Int {
        var result = mediumId
        result = 31 * result + tocCount
        return result
    }

    override fun toString(): String {
        return "RoomTocStat{" +
                "mediumId=" + mediumId +
                ", tocCount=" + tocCount +
                '}'
    }
}